### DISCUSSION TITLE
Nerd Fonts Considered Harmful

### DISCUSSION THREAD

**Discussion by ctrlcctrlv** [ryanoasis/nerd-fonts#1141] [Ideas]
I am writing this to inform that I recently wrote [this article](https://gist.github.com/ctrlcctrlv/4ae789dabaea774548da5f00a6bf16e9) and am working on patching `kitty` such that Nerd Fonts will no longer be necessary. Well, perhaps not exactly. I'm going to be patching `kitty` such that the “nerd fonts” that are necessary no longer fit the definition at present: fonts containing _only the icons_, sent by _the terminal application that needs them_. _Cf._ [kovidgoyal/kitty№2259](https://github.com/kovidgoyal/kitty/issues/2259#issuecomment-1460193065).

**Comment by Finii** [comment]
Hej @ctrlcctrlv 

Thanks for pointing to your article.

I believe some of the aspects of your text are true, and are in fact the same I say all the time. Some other aspects I do not concede.
Furthermore you judge Nerd Fonts only from a terminal emulator perspective, but in fact a considerable fraction of Nerd Fonts users use it in GUI contexts.

### Use SymbolsOnly font

Nerd Fonts also suggest NOT to use the patched fonts [link](https://github.com/ryanoasis/nerd-fonts/wiki/Font-Substitution-&-Fallback).

![image](https://user-images.githubusercontent.com/16012374/224254277-f6123406-e895-492e-98eb-83d4856e1365.png)

That does not even need a specific terminal application support, it just needs an application that follows normal font processing strategies - apparently kitty does not?

And well, the SymbolsOnly font are there, in the Release sets. Maybe not promoted as much as you would like.

### Old or missing fonts

Self patching has been improved a lot, and I guess most Nerds are easily able to patch whichever font they like.
As for sourcefont updates, although this has been true in the past all fonts will be updated this month, and afterwards we will start work to automatically create new patched versions if the source updates. This has been implemented for Cascadia Code (via Delugia; it auto updates within hours of the upstream update), but needs some tweeks with the Nerd Fonts release cycle (which is the hard part).

### Choosen codepoints

> McIntyre's chosen PUA space¹ often collides with your glyphs

I have never heard any complaint in this direction (except the glyphs at codepoints `FB00..FFFF` which fall into several Arabic and CJK blocks; but these will also be dropped in the next release (shifted already to PUA1 #773).

### Cross project Issues

>  people complain about the Nerd Fonts Project's version

Well. People also complain at Nerd Fonts about issues in the source font (which I usually fix upstream or at least re-issue there). And people complain at terminal applications about font issues, and they complain at font projects (e.g. Nerd Font) about terminal applications' problems. This is just how it is, users being no experts.

### Kitty receives fonts

I can not imagine that this is really a solution for many. If you have an application like `exa` or whatever you want that all users can use it. Only a very small fraction of users use `kitty`. If I would write an application I would think twice if I put the effort and maintenance time into a feature that most of my users can not use anyhow. Of course it is a nice gimmik, but how many complains will you get that the super font send feature does not work at some user, because they use Windows Terminal or iTerm or Tilix?

Related
* #829


### Patching is poor

It had a lot issues in the past, but that improved. Related is that SymbolsOnly (i.e. **not** specifically patched fonts) **also** has downsides. You are at the mercy of the terminal applications glyph scaling algorithm, which is ... unintelligent in some terminals.

### General tone

What you write sounds very harsh. Almost like bashing Nerd Fonts. Maybe that is unintended.
For example do I write an article about how useless `fontforge` is, because it destroys the `lowestRecPPEM` in fonts you open and export? It always write that value out with 9 - there is no possibility to change that behavior. No, I just fix it (the wrong values).

---

I can not really discern what your intentions are here, but I hope my comments are somewhat useful.

The kitty extension you propose does sound nice in any case.

But does it require to bash Nerd Fonts and/or express disgust ("I'd rather get rid of nerd fonts.").

"kitty, the best terminal emulator currently available" - certainly NOT. It does not even have a graphical settings dialog :roll_eyes: Choosing fonts is a chore. Shift-Ctrl-F2, they must be kidding, this is not 1984?

**Reply by mkanet** [reply]
@ctrlcctrlv I doubt you would ever be able to convince anyone to give up MS Terminal + Nerd Fonts + Oh My Posh... especially for Kitty.

**Reply by ctrlcctrlv** [reply]
It doesn't matter if I can or not. Kitty's APIs work their way into other terminals historically and will continue to do so. Konsole implements our graphics API. I don't care about non-free software so don't use MS anything so can't comment but it's an open standard which they are free to implement. I'm working on a full proposal still.
